### PR TITLE
267988: Added tier 2 MI for Postgres server Admin 

### PR DESCRIPTION
**Pull request by jeevankuduvaravindran**
# **What this PR does / why we need it**:
This PR contains the changes to create a tier 2 managed identity and set it as admin on postgres flexible server
[AB#267988](https://dev.azure.com/defragovuk/93c65135-d16b-49d6-a191-4a5313532779/_workitems/edit/267988)

# **Special notes for your reviewer**
*Any specific actions or notes on review?*

# Testing
*Any relevant testing information and pipeline runs.*

# Checklist (please delete before completing or setting auto-complete)
- [x] Story Work items associated (not Tasks)
- [ ] Successful testing run(s) link provided
- [x] Title pattern should be `{work item number}: {title}`
- [x] Description covers all the changes in the PR
- [ ] This PR contains documentation
- [ ] This PR contains tests

# **How does this PR make you feel**:
gif

### CODE REVIEW

File: infra/core/env/postgre-sql/flexible-server.bicep [hunk 1]
```diff
@@ -59,17 +74,41 @@ module flexibleServerDeployment 'br/SharedDefraRegistry:db-for-postgre-sql.flexi
...
+      }
+    ]
     configurations:[]
     delegatedSubnetResourceId : virtual_network::subnet.id
     privateDnsZoneArmResourceId: private_dns_zone.id
-    diagnosticWorkspaceId: ''
   }
 }
+
+var roleDefinitionId = 'acdd72a7-3385-48ef-bd42-f606fba81ae7'
```
Comment (DaRoszko) [inline infra/core/env/postgre-sql/flexible-server.bicep:95]: can you comment what this role name is in code?

File: infra/core/env/postgre-sql/flexible-server.json [hunk 2]
```diff
@@ -0,0 +1,1826 @@
+{
+  "$schema": "https://schema.management.azure.com/schemas/2019-04-01/deploymentTemplate.json#",
```
Comment (DaRoszko) [inline infra/core/env/postgre-sql/flexible-server.json:2]: a lot of json.. can this be bicep?

**Review by DaRoszko — CHANGES_REQUESTED** [review]

File: infra/core/env/postgre-sql/flexible-server.bicep (hunk 1 above)
Comment (jeevankuduvaravindran) [reply infra/core/env/postgre-sql/flexible-server.bicep]: @DaRoszko done

File: infra/core/env/postgre-sql/flexible-server.json (hunk 2 above)
Comment (jeevankuduvaravindran) [reply infra/core/env/postgre-sql/flexible-server.json]: @DaRoszko done

**Review by DaRoszko — APPROVED** [review]

**Review by kbabigumira — APPROVED** [review]

**Pull request merged**